### PR TITLE
Add LDAP variables to the configmap

### PR DESCRIPTION
**Pull request by zakkg3**
Include LDAP plugin variables to the config map.

I think we should push this to helm/charts official repo. I am a collaborator there, we can co-maintain this. 
let me know if it's ok for you we should push to stable, as I am using this and can say its stable. :)

cheers

to-do: put the password on a secret.

### CODE REVIEW

File: values.yaml [hunk 1]
```diff
@@ -72,7 +72,20 @@ taiga:
   # emailSmtpPort: 587
   # emailSmtpUser: "user@example.com"
   # emailSmtpPassword: ""
-
+  ldap:
+    Enabled: False
```
Comment (mvitale1989) [inline values.yaml]: The rest of the values.yaml file has `enabled` as fully lowercase, so it'd be nicer to make it more uniform
```suggestion
    enabled: False
```

File: templates/configmap.yaml [hunk 2]
```diff
@@ -19,3 +20,18 @@ data:
   TAIGA_EMAIL_PORT: {{ default 587 .Values.taiga.emailSmtpPort | quote }}
   TAIGA_EMAIL_USER: {{ required "taiga.emailSmtpUser must be set" .Values.taiga.emailSmtpUser | quote }}
 {{- end }}
+{{- if .Values.taiga.ldap.Enabled }}
```
Comment (mvitale1989) [inline templates/configmap.yaml]: Same as below, lowercase would be more fitting
```suggestion
{{- if .Values.taiga.ldap.enabled }}
```

File: templates/configmap.yaml [hunk 3]
```diff
@@ -19,3 +20,18 @@ data:
...
+  LDAP_AUTH_ENABLE: "True"
+  LDAP_START_TLS: {{ .Values.taiga.ldap.use_tls | default "False" |  quote }}
+  LDAP_SERVER: {{ .Values.taiga.ldap.ldap_server | quote }}
+  LDAP_PORT: {{ .Values.taiga.ldap.port | default 389 | quote }}
+  LDAP_BIND_DN: {{ .Values.taiga.ldap.bind_dn | default "cn=diegoarmandomaradona" |  quote }}
```
Comment (mvitale1989) [inline templates/configmap.yaml]: I like this default but it'd make more sense to leave it blank if not specified :smile: 
```suggestion
  LDAP_BIND_DN: {{ .Values.taiga.ldap.bind_dn | default "" |  quote }}
```

File: values.yaml [hunk 4]
```diff
@@ -72,7 +72,20 @@ taiga:
...
+    port: 389
+    bind_dn: "cn=something,ou=admins,ou=theouhere,ou=auth,o=ethz,c=ch"
+    base_dn: "basednhere"
+    password: "changeme"
+    username_attribute: "uid"
+    email_attribute: "mail"
+    full_name_attribute: "cn"
+    fallback: "normal"
+    save_password: False
+    filter: "(&(objectClass=posixAccount)(ou=some-ldap-group))"
```
Comment (mvitale1989) [inline values.yaml]: I'd put this to the image default, to minimize user surprise
```suggestion
    filter: "(mail=*)"
```

File: values.yaml [hunk 5]
```diff
@@ -72,7 +72,20 @@ taiga:
...
+    Enabled: False
+    use_tls: False
+    ldap_server: ldap://<ldap_server_host_name>
+    port: 389
+    bind_dn: "cn=something,ou=admins,ou=theouhere,ou=auth,o=ethz,c=ch"
+    base_dn: "basednhere"
+    password: "changeme"
+    username_attribute: "uid"
+    email_attribute: "mail"
+    full_name_attribute: "cn"
```
Comment (mvitale1989) [inline values.yaml]: Also here would be nice to default to the image and plugin default
```suggestion
    full_name_attribute: "displayName"
```

File: templates/configmap.yaml [hunk 6]
```diff
@@ -19,3 +20,18 @@ data:
...
+  LDAP_SEARCH_BASE: {{ .Values.taiga.ldap.base_dn |  quote }}
+  LDAP_BIND_PASSWORD: {{ .Values.taiga.ldap.password |  quote }}
+  LDAP_USERNAME_ATTRIBUTE: {{ .Values.taiga.ldap.username_attribute | default "uid" |  quote }}
+  LDAP_EMAIL_ATTRIBUTE: {{ .Values.taiga.ldap.email_attribute | default "mail" |  quote }}
+  LDAP_FULL_NAME_ATTRIBUTE: {{ .Values.taiga.ldap.full_name_attribute | default "cn" |  quote }}
```
Comment (mvitale1989) [inline templates/configmap.yaml]:
```suggestion
  LDAP_FULL_NAME_ATTRIBUTE: {{ .Values.taiga.ldap.full_name_attribute | default "displayName" |  quote }}
```

File: templates/configmap.yaml [hunk 7]
```diff
@@ -19,3 +20,18 @@ data:
...
+  LDAP_FALLBACK: {{ .Values.taiga.ldap.fallback | default "normal" |  quote }}
+  LDAP_SAVE_LOGIN_PASSWORD: {{ .Values.taiga.ldap.save_password | default "False" |  quote }}
+  LDAP_SEARCH_FILTER_ADDITIONAL: {{ .Values.taiga.ldap.filter }}
```
Comment (mvitale1989) [inline templates/configmap.yaml]:
```suggestion
  LDAP_SEARCH_FILTER_ADDITIONAL: {{ .Values.taiga.ldap.filter | default "" | quote }}
```

File: templates/configmap.yaml [hunk 8]
```diff
@@ -19,3 +20,18 @@ data:
...
   TAIGA_EMAIL_USER: {{ required "taiga.emailSmtpUser must be set" .Values.taiga.emailSmtpUser | quote }}
 {{- end }}
+{{- if .Values.taiga.ldap.Enabled }}
+  LDAP_AUTH_ENABLE: "True"
+  LDAP_START_TLS: {{ .Values.taiga.ldap.use_tls | default "False" |  quote }}
+  LDAP_SERVER: {{ .Values.taiga.ldap.ldap_server | quote }}
+  LDAP_PORT: {{ .Values.taiga.ldap.port | default 389 | quote }}
+  LDAP_BIND_DN: {{ .Values.taiga.ldap.bind_dn | default "cn=diegoarmandomaradona" |  quote }}
+  LDAP_SEARCH_BASE: {{ .Values.taiga.ldap.base_dn |  quote }}
+  LDAP_BIND_PASSWORD: {{ .Values.taiga.ldap.password |  quote }}
```
Comment (mvitale1989) [inline templates/configmap.yaml:30]: As you suggested, this should go to secret.yaml

**Review by mvitale1989 — COMMENTED** [review]
Hello and thanks for your contribution! Sure, it'd be great to see this merged actually. Shall i re-submit a PR to the helm/charts repo?
Left a couple of suggestions on the MR. We could also bump the default image to `20191031-4.2.14`, which is the latest i built for the 4.x version.

File: templates/configmap.yaml (hunk 3 above)
Comment (zakkg3) [reply templates/configmap.yaml]: :D completely forgot about this LOL

File: templates/configmap.yaml (hunk 8 above)
Comment (zakkg3) [reply templates/configmap.yaml:30]: Yap. I say: let's merge this and when I have time I will create another pr for this., :)

Comment (mvitale1989) [reply templates/configmap.yaml:30]: Yep sounds fine

**Pull request merged**